src/tracked_struct/struct_map.rs

```diff
@@ -201,6 +209,50 @@ where
         unsafe { C::struct_from_raw(data.as_raw()) }
     }
 
+    fn get_and_validate_last_changed<'db>(
```

Comment:
I would find comment describing how it is different from `get_from_map` useful because the difference is very subtle

Reply:
Added the comment.

One of my questions here for @nikomatsakis is whether instead of adding a new `get_and_validate_last_changed`, maybe the implementation of `get_from_map` should just always do this. It's possible that other code paths that just call `.get` are still susceptible to this bug?

But it's also a little weird if a method called `get` can mutate the stored data.